exchanges/state/state.go

```diff
@@ -0,0 +1,392 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/thrasher-corp/gocryptotrader/common/convert"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/alert"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+var (
+	manager Manager
```

Comment:
Is there a way we can do this without globals? I would really prefer it. It looks like the manager aspect of this package could really all be moved to the engine level manager and this package still works. You're mostly using this manager as a pointer at the engine level anyway

Reply:
Yeah the reason for this is to keep in line with the orderbook/ticker packages so if somebody uses exchange package(s) decoupled from the engine you can still update, retrieve and implement easily.

Reply:
I don't think the manager here needs to have knowledge of all exchange currency states in this case. I think an exchange can keep track of its own `*States` property, in the same way each exchange retrieves and updates what currencies are available to itself. 

Someone using package level exchanges would still have access to the `States` property of the `exchange.Base` and call `UpdateCurrencyStates`. Then the engine level `CurrencyStateManager` is responsible for asking each exchange to update their states on a timely basis.

If you remove the `manager` global var and the `Manager` type from `/exchanges/currencystate/currency_state.go`, you can do the following:
```
func SetupCurrencyState() *States {
	return &States{m: make(map[asset.Item]map[*currency.Item]*Currency)}
}
```
Then in `/exchanges/exchange.go` function `SetupDefaults(...)` becomes
```
b.States = currencystate.SetupCurrencyState()
```
Finally, in `/engine/currency_state_manager.go` you can remove the `*currencystate.Manager` property and everything is still tracked and working because the `order_manager` for example checks if the currency can be traded via an exchange interface call. If things were checked against the `/engine/currency_state_manager.go`, then that could be different, but at this current point it is not.

I think it can simplify things a bit while still addressing your concerns

Reply:
Thanks for explaining that, yeah I see what your saying, I will follow your example.